pos_gift_card_design/models/pos_order.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+import base64
+
+
+class PosOrder(models.Model):
+    _inherit = "pos.order"
+
+    @api.model
+    def create_from_ui(self, orders, draft=False):
+        order_ids = super(PosOrder,self).create_from_ui(orders, draft)
+        for order in self.sudo().browse([o["id"] for o in order_ids]):
+            gift_card_config = order.config_id.gift_card_settings
+            for line in order.lines:
+                # Change condition and check if the object is a gift card to create gift card and not only when the object is the configured gift card
```

Comment:
it could be nice to at least test this case in unit test where used gift card is not the same as the once configured in ps config